### PR TITLE
fix #1

### LINKED ISSUE #1: efficient looping
The statement
```python
diff_results = self.__getLineNumbers(element.a.get('href'))
```
gets executed 2 times inside the loop in `def __extractInfo(self, url):` method.

The problem is both times the link is same, so it should run only 1 time(i.e that is one iteration of inner loop is redundant)

### PR DESCRIPTION
**Pull request by vhsw**
Sorry, I went a little bit crazy refactoring this. But you can get an idea of how to fix the bug described in #1

### CODE REVIEW

**Comment by Bhupesh-V** [comment]
@vhsw  your PR helped me a lot :rocket: 
But I can't merge these changes because you are using Python 3.8 and I  have developed this in Python 3.6 so the type hints won't work or will they :confused:

**Comment by vhsw** [comment]
Type hints were added in Python 3.5, so this code should work fine in Python 3.6
I added [Travis CI](https://travis-ci.org/vhsw/PlagCheck/) with some tests to make sure everything works

**Review by Bhupesh-V — CHANGES_REQUESTED** [review]
Removing the following lines from `deno.py` will give `USER_ID` error
```python
from dotenv import load_dotenv
load_dotenv()
````

Also are you using `pytest` for running tests ?

File: plagcheck/plagcheck_test.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+"""Tests for the MOSS interface package for CodeClassroom"""
+from plagcheck import plagcheck
```
Comment (Bhupesh-V) [inline plagcheck/plagcheck_test.py:2]:
```suggestion
import plagcheck
```

File: demo.py [hunk 2]
```diff
@@ -1,17 +1,14 @@
+"""Usage example"""
 import os
-import PlagCheck.check as plagcheck
-from dotenv import load_dotenv
-load_dotenv()
+import pprint
+from plagcheck import plagcheck
```
Comment (Bhupesh-V) [inline demo.py:4]:
```suggestion
from plagcheck import plagcheck
from dotenv import load_dotenv
load_dotenv()
```

**Comment by Bhupesh-V** [comment]
> Type hints were added in Python 3.5, so this code should work fine in Python 3.6

Oh ! ok wasn't aware of that, thanks :+1:

**Comment by Bhupesh-V** [comment]
@all-contributors please add @vhsw for code, bug, test

**Pull request merged**